File: 2016-08-15-WebViewExample/src/main/java/tech/thdev/webviewexample/view/main/presenter/MainPresenter.java
package tech.thdev.webviewexample.view.main.presenter;

import org.jetbrains.annotations.NotNull;

import tech.thdev.kotlin_example_01.base.presenter.AbstractPresenter;
import tech.thdev.webviewjavascriptinterface.util.StringUtilKt;

/**
 * Created by tae-hwan on 8/14/16.
 */

public class MainPresenter extends AbstractPresenter<MainContract.View> implements MainContract.Presenter {

    private static final String BLANK_PAGE = "about:blank";

    @Override
    public void attachView(@NotNull MainContract.View view) {
        super.attachView(view);

        view.onPresenter(this);
    }

    @Override
    public void defaultLoadUrl() {
        if (isAttachView()) {
            getView().loadUrl(BLANK_PAGE);
        }
    }

    @Override
    public void validationUrl(String text) {
        String url = StringUtilKt.checkUrlText(text);

        if (isAttachView()) {
            getView().loadUrl(url);
        }
    }

    @Override
    public void onFinish(String url) {
        if (BLANK_PAGE.equals(url)) {
            url = "";
        }

        if (isAttachView())
            getView().setUrl(url);
    }

    @Override
    public void onUrlChange(String url) {
        if (isAttachView())
            getView().setUrl(url);
    }

    @Override
    public void onScroll(int l, int t, int oldl, int oldt) {
        boolean isTop = t < oldt;
        if (isAttachView()) {
            getView().webViewScrollChanged(isTop);
        }
    }

    @Override
    public void onProgressChanged(int newProgress) {
        if (isAttachView()) {
            getView().webViewProgressChanged(newProgress);
        }
    }
}

File: app/src/main/java/tech/thdev/app/ui/main/presenter/MainPresenter.kt
package tech.thdev.app.ui.main.presenter

import tech.thdev.app.base.presenter.AbstractPresenter
import tech.thdev.app.util.checkUrlText

/**
 * Created by tae-hwan on 8/14/16.
 */
class MainPresenter : AbstractPresenter<MainContract.View>(), MainContract.Presenter {

    companion object {
        private const val BLANK_PAGE = "about:blank"
    }

    override fun attachView(view: MainContract.View) {
        super.attachView(view)
        view.onPresenter(this)
    }

    override fun defaultLoadUrl() {
        view?.loadUrl(BLANK_PAGE)
    }

    override fun validationUrl(text: String?) {
        val url: String = text.checkUrlText()
        view?.loadUrl(url)
    }

    override fun onFinish(url: String?) {
        view?.setUrl(url)
    }

    override fun onUrlChange(url: String?) {
        view?.setUrl(url)
    }

    override fun onScroll(l: Int, t: Int, oldl: Int, oldt: Int) {
        val isTop = t < oldt
        view?.webViewScrollChanged(isTop)
    }

    override fun onProgressChanged(newProgress: Int) {
        view?.webViewProgressChanged(newProgress)
    }
}